sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/HintErrorLogger.scala

```diff
@@ -42,6 +45,17 @@ object HintErrorLogger extends HintErrorHandler with Logging {
     logWarning(s"A join hint $hint is specified but it is not part of a join relation.")
   }
 
+  override def joinBuildSideNotSupported(joinType: JoinType, joinHint: JoinHint): Unit = {
```

Comment:
cc @maryannxue

Reply:
can we make the name more general like `hintNotSupported`?

Reply:
e.g.
```
def hintNotSupported(hint: HintInfo, reason: String): Unit = {
  logWarning("Hint $hint is not supported in the query: " + reason)
}
```

Reply:
updated